src/main/java/com/igot/cb/consentacknowledge/ConsentAcknowledgeServiceImpl.java

```
@@ -0,0 +1,155 @@
+        if (StringUtils.isEmpty(userId)) {
+            return response;
+        }
+        String errMsg = validateAcknowledgeDeclarationRequest(requestDataMap);
+        if (!StringUtils.isEmpty(errMsg)) {
+            ProjectUtil.errorResponse(response, errMsg, HttpStatus.BAD_REQUEST);
+            return response;
+        }
+        String contentId = (String) requestDataMap.get(Constants.CONTENT_ID);
+        String consentId = (String) requestDataMap.get(Constants.CONSENT_ID);
+        Map<String, Object> additionalData = (Map<String, Object>) requestDataMap.get(Constants.ADDITIONAL_ATTRIBUTES);
+        String additionalDataStr = null;
+        try {
+            additionalDataStr = objectMapper.writeValueAsString(additionalData);
+        } catch (JsonProcessingException e) {
+            logger.error("Error while converting additionalData to string", e);
+            ProjectUtil.errorResponse(response,"Failed to Parse the additional attributes", HttpStatus.BAD_REQUEST);
+            return response;
+        }
+        Map<String, Object> compositeKeyMap = new HashMap<>();
+        compositeKeyMap.put(Constants.USER_ID, userId);
+        compositeKeyMap.put(Constants.CONSENT_ID, consentId);
+        Map<String, Object> declarationDataMap = new HashMap<>();
+        ZoneId zoneId = ZoneId.of("Asia/Kolkata");
+        ZonedDateTime submittedAt = ZonedDateTime.ofInstant(Instant.now(), zoneId);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS");
+        declarationDataMap.put(Constants.SUBMITTED_BY, userId);
+        declarationDataMap.put(Constants.ADDITIONAL_ATTRIBUTES, additionalDataStr);
+        declarationDataMap.put(Constants.SUBMITTED_AT, submittedAt.format(formatter));
+        try {
+            cassandraOperation.insertRecord(Constants.KEYSPACE_SUNBIRD, Constants.TABLE_DECLARATION_ACKNOWLEDGMENT, Constants.CONTENT_ID, contentId, compositeKeyMap, declarationDataMap);
```

when there is an exception -- insertRecord doesn't re-throw.
either you need to re-throw the exception or validate the return response.

---

Validated the return response:

response = (ApiResponse) cassandraOperation.insertRecord(Constants.KEYSPACE_SUNBIRD, Constants.TABLE_DECLARATION_ACKNOWLEDGMENT, Constants.CONTENT_ID, contentId, compositeKeyMap, declarationDataMap);
            if (Constants.FAILED.equals(response.get(Constants.RESPONSE))) {
                logger.error("Error while inserting declaration acknowledgment record in DB: {}",
                        response.get(Constants.ERROR_MESSAGE));
                ProjectUtil.errorResponse(
                        response,
                        "Failed to acknowledge declaration. Please try again later.",
                        HttpStatus.INTERNAL_SERVER_ERROR
                );
                return response;
            }